fixtures-generator/json2sql.py

```diff
@@ -0,0 +1,66 @@
+import json_helpers
+
+import argparse
+import datetime
+import os
+
+
+def hasura_timestamp():
+    epoch = datetime.datetime.utcfromtimestamp(0)
+    now = datetime.datetime.now()
+    return round((now - epoch).total_seconds() * 1000)
+
+
+def generate_up_migration(json_data, output_folder, migration_name, write_mode):
+    migration_file = os.path.join(output_folder, migration_name + '.up.sql')
+    with open(migration_file, write_mode) as output_file:
+        for table, data in json_data.items():
+            for data_item in data:
+                columns = ','.join(data_item.keys())
+                values = ','.join(
+                    map(lambda val: f"'{val}'", data_item.values()))
+                output_file.write(
+                    f'INSERT INTO public.{table} ({columns}) VALUES ({values});\n')
+
+
+def generate_down_migration(json_data, output_folder, migration_name, write_mode):
+    migration_file = os.path.join(output_folder, migration_name + '.down.sql')
+    with open(migration_file, write_mode) as output_file:
+        for table, data in reversed(json_data.items()):
+            for data_item in reversed(data):
+                condition = []
+                for col, val in reversed(data_item.items()):
+                    condition.append(f'{col}=\'{val}\'')
+                where = ' AND '.join(condition)
+                output_file.write(f'DELETE FROM {table} WHERE {where};\n')
```

Comment:
That is the right way of performing the down migration.

Reply:
Yes, this only removes the data added by the migration. I was deleting the whole content. I must admit this is way better.

Reply:
No, you were not removing the whole content. You were removing the whole content of the last table.